include/boost/geometry/formulas/karney_direct.hpp

```diff
@@ -0,0 +1,290 @@
+        CT const cos_sigma12 = cos(sigma12);
+
+        CT const sin_sigma2
+            = sin_sigma1 * cos_sigma12 + cos_sigma1 * sin_sigma12;
+        CT const cos_sigma2
+            = cos_sigma1 * cos_sigma12 - sin_sigma1 * sin_sigma12;
+
+        if (BOOST_GEOMETRY_CONDITION(CalcRevAzimuth))
+        {
+            CT const sin_alpha2 = sin_alpha0;
+            CT const cos_alpha2 = cos_alpha0 * cos_sigma2;
+
+            result.reverse_azimuth = atan2(sin_alpha2, cos_alpha2);
+
+            // Convert the angle to radians.
+            result.reverse_azimuth /= math::d2r<T>();
+        }
+
+        if (BOOST_GEOMETRY_CONDITION(CalcCoordinates))
+        {
+            // Find the latitude at the second point.
+            CT const sin_beta2
+                = cos_alpha0 * sin_sigma2;
+            CT const cos_beta2
+                = boost::math::hypot(sin_alpha0, cos_alpha0 * cos_sigma2);
+
+            result.lat2 = atan2(sin_beta2, one_minus_f * cos_beta2);
+
+            // Convert the coordinate to radians.
+            result.lat2 /= math::d2r<T>();
+
+            // Find the longitude at the second point.
+            CT const sin_omega2 = sin_alpha0 * sin_sigma2;
+            CT const cos_omega2 = cos_sigma2;
+
+            CT const omega12 = atan2(sin_omega2 * cos_omega1 - cos_omega2 * sin_omega1,
+                                     cos_omega2 * cos_omega1 + sin_omega2 * sin_omega1);
+
+            CT coeffs_A3[SeriesOrder];
+            series_expansion::evaluate_coeffs_A3<double, SeriesOrder>(n, coeffs_A3);
+
+            CT const A3 = math::horner_evaluate(epsilon, coeffs_A3, coeffs_A3 + SeriesOrder);
+            CT const A3c = -f * sin_alpha0 * A3;
+
+            // Compute the size of coefficient array.
+            size_t const coeffs_C3_size = (SeriesOrder * (SeriesOrder - 1)) / 2;
+            CT coeffs_C3x[coeffs_C3_size];
+            series_expansion::evaluate_coeffs_C3x<double, SeriesOrder>(n, coeffs_C3x);
+
+            // Evaluate C3 coefficients.
+            CT coeffs_C3[SeriesOrder];
+            series_expansion::evaluate_coeffs_C3<double, SeriesOrder>(epsilon, coeffs_C3, coeffs_C3x);
+
+            CT const B31 = sin_cos_series(sin_sigma1, cos_sigma1, coeffs_C3);
+
+            CT const lam12 = omega12 + A3c *
+                             (sigma12 + (sin_cos_series(sin_sigma2,
+                                                        cos_sigma2,
+                                                        coeffs_C3) - B31));
+
+            // Convert to radians to get the
+            // longitudinal difference.
+            CT const lon12 = lam12 / math::d2r<T>();
+
+            // Add the longitude at first point to the longitudinal
+            // difference and normalize the result.
+            result.lon2 = math::normalize_angle(math::normalize_angle(lon1) +
+                                                math::normalize_angle(lon12));
+        }
+
+        if (BOOST_GEOMETRY_CONDITION(CalcQuantities))
```

Comment:
this part of code duplicates code from https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/formulas/differential_quantities.hpp is it possible to integrate ?

Reply:
I actually tried using that code. However, almost all the test cases fail if I do. I find that odd, since it seems to be working well with Vincenty and Thomas. I will try and dig a little deeper into the code and see if I'm missing out on a step.